main_board/app/src/optics/ir_camera_system/ir_camera_timer_settings.h

```diff
@@ -2,11 +2,36 @@
 
 #include <errors.h>
 #include <stdint.h>
+#ifndef CONFIG_ZTEST
+#include <zephyr/devicetree.h>
+#endif
 
+#if defined(CONFIG_BOARD_PEARL_MAIN)
 #define IR_CAMERA_SYSTEM_MAX_IR_LED_ON_TIME_US 5000
-#define IR_CAMERA_SYSTEM_MAX_FPS               60
-#define ASSUMED_TIMER_CLOCK_FREQ_MHZ           170
-#define ASSUMED_TIMER_CLOCK_FREQ               (ASSUMED_TIMER_CLOCK_FREQ_MHZ * 1000000)
+#define IR_CAMERA_SYSTEM_MAX_IR_LED_DUTY_CYCLE 0.15
+#else
+#define IR_CAMERA_SYSTEM_MAX_IR_LED_ON_TIME_US 8000
+#define IR_CAMERA_SYSTEM_MAX_IR_LED_DUTY_CYCLE 0.25
+#endif
+#define IR_CAMERA_SYSTEM_MAX_FPS 60
+#if defined(CONFIG_ZTEST)
+#define TIMER_CLOCK_FREQ_MHZ 170L
+#define TIMER_CLOCK_FREQ_HZ  (TIMER_CLOCK_FREQ_MHZ * 1000000)
+#else
+#define TIMER_CLOCK_FREQ_HZ                                                    \
+    (DT_PROP(DT_NODELABEL(rcc), clock_frequency) /                             \
+     DT_PROP(DT_NODELABEL(rcc), ahb_prescaler) /                               \
+     DT_PROP(DT_NODELABEL(rcc), apb1_prescaler))
```

Comment:
I think you can use `DT_PROP_OR` to give a default value in case the node is not defined

Reply:
done

Reply:
doesnt work as I requires including <zephyr/devicetree.h> which includes <devicetree_generated.h> which doesn`t exist when no devicetree is compiled